ui/imports/shared/views/chat/ProfileContextMenu.qml

```diff
@@ -1,297 +1,215 @@
-    }
+    property bool ensVerified: false
+    property int onlineStatus: Constants.onlineStatus.unknown
 
-    readonly property bool userTrustIsUnknown: contactDetails && contactDetails.trustStatus === Constants.trustStatus.unknown
-    readonly property bool userIsUntrustworthy: contactDetails && contactDetails.trustStatus === Constants.trustStatus.untrustworthy
-    readonly property bool userIsLocallyTrusted: contactDetails && contactDetails.trustStatus === Constants.trustStatus.trusted
+    property bool hasLocalNickname: false
 
     signal openProfileClicked(string publicKey)
     signal createOneToOneChat(string communityId, string chatId, string ensName)
-
-    onClosed: {
-        // Reset selectedUserPublicKey so that associated properties get recalculated on re-open
-        selectedUserPublicKey = ""
-    }
-
-    ProfileHeader {
-        width: parent.width
-        height: visible ? implicitHeight : 0
-
-        displayNameVisible: false
-        displayNamePlusIconsVisible: true
-        editButtonVisible: false
-        displayName: StatusQUtils.Emoji.parse(root.selectedUserDisplayName, StatusQUtils.Emoji.size.verySmall)
-        pubkey: root.selectedUserPublicKey
-        icon: root.selectedUserIcon
-        trustStatus: contactDetails && contactDetails.trustStatus ? contactDetails.trustStatus
-                                                                  : Constants.trustStatus.unknown
-        Binding on onlineStatus {
-            value: contactDetails.onlineStatus
-            when: !root.isMe
+    signal reviewContactRequest(string publicKey)
+    signal sendContactRequest(string publicKey)
+    signal editNickname(string publicKey)
+    signal removeNickname(string publicKey, string displayName)
+    signal unblockContact(string publicKey)
+    signal markAsUntrusted(string publicKey)
+    signal removeTrustStatus(string publicKey)
+    signal removeContact(string publicKey)
+    signal blockContact(string publicKey)
+    signal opened()
+    signal closed()
+
+    Loader {
+        id: menuLoader
+        sourceComponent: {
+            switch (root.profileType) {
+                case Constants.profileType.self:
+                    return selfMenuComponent
+                case Constants.profileType.bridged:
+                    return bridgeMenuComponent
+                case Constants.profileType.blocked:
+                    return blockedMenuComponent
+                default:
+                    return regularMenuComponent
+            }
         }
-        isContact: root.isContact
-        isBlocked: root.isBlockedContact
-        isCurrentUser: root.isMe
-        userIsEnsVerified: (!!contactDetails && contactDetails.ensVerified) || false
-        isBridgedAccount: root.isBridgedAccount
-    }
-
-    StatusMenuSeparator {
-        topPadding: root.topPadding
-        visible: !root.isBridgedAccount
-    }
+        onLoaded: {
```

Comment:
I'd do it the other way: Each `component` to bind its properties and signals to the `root` properties and signals.

The are a few advantages: 
1. we won't need this imperative approach and all the bindings would work as expected.
2. No need to create most of the the `if` branches
3. No need for the manual connections
4. Failing fast in case of type errors at qml compilation time
And probably more.

In general it's a good approach to choose the declarative approach as it brings many advantages.